simulation/utils/generate_map.py

```diff
@@ -143,54 +164,51 @@ class Difficulty(Enum):
     REALLY_HARD = 4
     WTF = 5
 
-# ************
-def intersect(A, B, C, D):
-    """Helper of is_crossing function
-    Returns True if line segments AB and CD intersect"""
-    ccw = lambda A, B, C: (C.y - A.y) * (B.x - A.x) > (B.y - A.y) * (C.x - A.x)
-    return ccw(A, C, D) != ccw(B, C, D) and ccw(A, B, C) != ccw(A, B, D)
-
-# ************
-def next_point(prev_point, step, alpha, y_res):
+    
+def generate_next_point(prev_point, step, alpha, y_resolution):
     """Function returning next random point for the generalized map"""
     y_range = 2 * step * math.tan(alpha / 2)
-    ymin = prev_point.y - y_range / 2
-    ymin = ymin if ymin >= 0 else 0
-    ymax = prev_point.y + y_range / 2
-    ymax = ymax if ymax < y_res - 20 else y_res - 20
+
+    y_min = prev_point.y - y_range / 2
+    y_min = y_min if y_min >= 0 else 0
+    y_max = prev_point.y + y_range / 2
+    y_max = y_max if y_max < y_resolution - Y_CEILING_LIMIT else y_resolution - Y_CEILING_LIMIT
+
     p = random.random()
     if p >= 0.3:
         x = prev_point.x + step
     else:
         x = prev_point.x - step if prev_point.x > step else prev_point.x + step
-    y = random.uniform(ymin, ymax)
-    point = Point(x, y)
-    return point
 
-# ************
-def prepare_map(resolution, step, angle_range, x_offset, y_offset, seed):
+    y = random.uniform(y_min, y_max)
+
+    next_point = Point(x, y)
+
+    return next_point
+
+
+def prepare_map_before_interpolation(resolution, step, angle_range, x_offset, y_offset, seed):
     """Function that prepares real map - interpolates random points generated by function next_point
         returns interpolated and smoothed map in X and Y arrays
     """
     x_res = resolution[0]
     y_res = resolution[1]
 
-    x = 0
-    y = random.randrange(y_res / 4, y_res * 3 / 5) if not y_offset else y_offset
-    point = Point(x, y)
-    game_map = Map(point, x_offset, resolution, seed)
+    y_offset = random.randrange(y_res / 4, y_res * 3 / 5) if not y_offset else y_offset
+    point = Point(x_offset, y_offset)
+    game_map = Map(starting_point=point, resolution=resolution, seed=seed)
```

Comment:
Nie jestem pewien czy jak zmienna nazywa się tak samo jak argument do którego ją przypisujesz to czy jest sens pisać oba (np. seed = seed), ale to już indywidualna kwestia ;)

Reply:
Ano normalnie tak nie robię, ale jak zobaczyłam jak to u Was wygląda z tym przekazywaniem argumentów do funkcji tak dokładnie to uznałam, że będę się tego trzymać xd
Ale spoko, moge to zmienić

Reply:
Nie musisz - to już zupełnie zależy od Ciebie. Na pewno przekazuje to jakąś informację, także na 100% nie jest to błąd.